internal/hcsoci/resources_lcow.go

```diff
@@ -6,19 +6,33 @@ package hcsoci
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/Microsoft/hcsshim/internal/log"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
 )
 
 const lcowMountPathPrefix = "/mounts/m%d"
 const lcowGlobalMountPrefix = "/run/mounts/m%d"
+const lcowNvidiaMountPath = "/run/nvidia"
+
+func getNvidiaGpuVhdPath(coi *createOptionsInternal) (string, error) {
+	gpuVhdPath, ok := coi.Spec.Annotations["io.microsoft.lcow.nvidiagpuvhdpath"]
+	if !ok || gpuVhdPath == "" {
+		// default path was not set in config file, switch to default
+		gpuVhdPath = filepath.Join(filepath.Dir(os.Args[0]), "gpu.vhd")
```

Comment:
I think we should name the default file `nvidiagpu.vhd`, in case we add similar VHDs for other vendors.